package.json

```diff
@@ -3,11 +3,16 @@
   "version": "0.1.0",
   "description": "asciinema GIF generator",
   "main": "main.js",
+  "bin": {
+    "asciicast2gif": "asciicast2gif"
```

Comment:
So this includes the shell script in the output npm package? I didn't know you can use non-node scripts as bins, that's cool!

Reply:
Yea all NPM does, is create a symlink. From here the system does it's thing. Since node is run from inside the script everything is cool. Looking back if you want to support Windows I can rewrite it as a js file if you like.

Reply:
No, I don't want to go to the Windows rabbit hole right now. This is fine. So this makes a symlink, but does it also ensure this shell script is uploaded as part of npm package? I assume when you publish to npm not whole of project source tree is uploaded :)

Reply:
Fair enough, you assume incorrectly :) NPM uploads/downloads the entire project except things defined in the .gitignore or .npmignore file:

> Note that everything in the directory will be included unless it is ignored by a local .gitignore or .npmignore file as described in [npm-developers](https://docs.npmjs.com/misc/developers).

[source](https://docs.npmjs.com/getting-started/publishing-npm-packages#publishing-the-package)

> .npmignore files follow the same pattern rules as [.gitignore](https://git-scm.com/book/en/v2/Git-Basics-Recording-Changes-to-the-Repository#Ignoring-Files) files:
>
>    Blank lines or lines starting with # are ignored.
    Standard glob patterns work.
    You can end patterns with a forward slash / to specify a directory.
    You can negate a pattern by starting it with an exclamation point !.
[source](https://docs.npmjs.com/misc/developers#keeping-files-out-of-your-package)